esmvalcore/experimental/recipe_output.py

```diff
@@ -150,6 +159,33 @@ def __init__(self, task_output: dict, session=None, info=None):
                 description=diagnostic_info.get('description'),
             )
 
+            # Add data to filters
+            for task in tasks:
+                for file in task.files:
+                    self._add_to_filters(file.attributes)
```

Comment:
During this loop, `self.filters` is a `Mapping[Str, Set]`, but then after the sorting, it becomes a `Mapping[Str, List]`. This suggests that during the loop it really is a temporary variable. Try to reflect that in the code.

Reply:
Good point, I didn't realize it was a bad practice to reuse the variable that way.

Reply:
Since this is all happening inside `__init__`, I would suggest to not use `self.filter` until the end, probably best introduce a function `_init_filter(...)` or similar and have in `__init__`: `self.filters = (self.)_init_filters(...)`.

Reply:
I've done it by creating a temporary variable assigning the value just at the end, and refactoring the auxiliary functions to class methods.